lib/pacemaker/libpacemaker_private.h

```diff
@@ -42,6 +40,27 @@ struct resource_alloc_functions_s {
                                pcmk__colocation_t *colocation,
                                bool for_dependent);
 
+    /*!
+     * \internal
+     * \brief Update nodes with scores of colocated resources' nodes
+     *
+     * Given a table of nodes and a resource, update the nodes' scores with the
+     * scores of the best nodes matching the attribute used for each of the
```

Comment:
Just the best nodes or all nodes? Sincere question; I thought all available nodes matching the attribute would factor into the update.

Reply:
add_node_scores_matching_attr() does the actual adding, and it uses best_node_score_matching_attr()